File: go/stocks/bank.go
package stocks

import "errors"

// Author:	DragonChen https://github.com/dragonchen-tw/
// Title:	bank package
// Date:	2022/09/01

type Bank struct {
	exchangeRates map[string]float64
}

func NewBank() Bank {
	return Bank{exchangeRates: make(map[string]float64)}
}

func (bank Bank) AddExchangeRate(fromCurrency, toCurrency string, rate float64) {
	key := fromCurrency + "->" + toCurrency
	bank.exchangeRates[key] = rate
}

func (bank Bank) Convert(money Money, currencyTo string) (*Money, error) {
	if money.currency == currencyTo {
		return &money, nil
	}

	key := money.currency + "->" + currencyTo
	rate, ok := bank.exchangeRates[key]

	if !ok {
		return nil, errors.New(key)
	}

	return &Money{money.amount * rate, currencyTo}, nil
}

File: py/bank.py
from money import Money

class Bank:
    def __init__(self):
        self.exchangeRates = {}
    
    def addExchangeRate(self, fromCurrency: str, toCurrency: str, rate: float) -> None:
        key = f'{fromCurrency}->{toCurrency}'
        self.exchangeRates[key] = rate
    
    def convert(self, money: Money, currency: str) -> Money:
        if money.currency == currency:
            return money
        
        key = f'{money.currency}->{currency}'
        if key not in self.exchangeRates:
            raise Exception(key)
        
        rate = self.exchangeRates[key]
        return Money(money.amount * rate, currency)